### ISSUE TITLE
Get rid of `NSSecureCodingOptionalKey`

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/Defaults#40]
I got rid of `OptionalKey` in https://github.com/sindresorhus/Defaults/commit/b2fdee2055c87149bef0b2fec0726a87c191926c, but I was not able to remove `NSSecureCodingOptionalKey` as it seems optionals cannot conform to `NSSecureCoding`, or maybe it has something to do with `NSSecureCoding` being an Objective-C protocol.

Try changing https://github.com/sindresorhus/Defaults/blob/77c05abe1001f7d6fd6076816a227012129089d2/Tests/DefaultsTests/DefaultsTests.swift#L459 to:

```swift
let key = Defaults.NSSecureCodingKey<ExamplePersistentHistory?>("observeNSSecureCodingOptionalKey")
```

and see the error.

Help wanted 🙏

**Comment by sindresorhus** [comment]
// @Ybrin

**Comment by sindresorhus** [comment]
Fixed by #57
